pylint/checkers/base.py

```diff
@@ -2593,6 +2594,41 @@ def _check_type_x_is_y(self, node, left, operator, right):
         self.add_message("unidiomatic-typecheck", node=node)
 
 
+class ThreadingChecker(_BasicChecker):
+    """Checks for threading module
+
+    - useless with lock - locking used in wrong way that has no effect (with threading.Lock():)
+    """
+
+    msgs = {
+        "W9990": (  # TODO
+            "Useless usage of with statement and %s()",
```

Comment:
```suggestion
            "'%s()' directly created in 'with' has no effect",
```
I think the formulation of the extended description is nice and could be used here too.

Reply:
Updated